docs/agents/GoDocs.md

```diff
@@ -5,179 +5,241 @@
+  ByBlockID(blockID flow.Identifier) (*flow.Header, error)
+  ```
 
 ### Parameters
-   - Only document parameters separately when they have non-obvious aspects:
-     - Complex constraints or requirements
-     - Special relationships with other parameters
-     - Formatting or validation rules
-     - Example:
-       ```go
-       // ValidateTransaction validates the transaction against the current state.
-       //
-       // Parameters:
-       //   - script: must be valid BPL-encoded script with max size of 64KB
-       //   - accounts: must contain at least one account with signing capability
-       ```
+- Only document parameters separately when they have non-obvious aspects:
+  - Complex constraints or requirements
+  - Special relationships with other parameters
+  - Formatting or validation rules
+  - Example:
+    ```go
+    // ValidateTransaction validates the transaction against the current state.
+    //
+    // Parameters:
+    //   - script: must be valid BPL-encoded script with max size of 64KB
+    //   - accounts: must contain at least one account with signing capability
+    ```
 
 ### Returns
-   - Only document return values if there is **additional information** necessary to interpret the function's or method's return values, which is not apparent from the method signature's return values
-   - When documenting non-error returns, be concise and focus only on non-obvious aspects:
-     ```go
-     // Example 1 - No return docs needed (self-explanatory):
-     // GetHeight returns the block's height.
+- Only document return values if there is **additional information** necessary to interpret the function's or method's return values, which is not apparent from the method signature's return values
+- When documenting non-error returns, be concise and focus only on non-obvious aspects:
+  Example 1 - No return docs needed (self-explanatory):
+  ```go
+  // GetHeight returns the block's height.
+  ```
+
+  Example 2 - Additional context needed within method description:
+  ```go
+  // GetPipeline returns the execution pipeline, or nil if not configured.
+  ```
+
+  Example 3 - Complex return value needs explanation:
+  ```go
+  // GetBlockStatus returns the block's current status.
+  //
+  // Returns:
+  //   - status: PENDING if still processing, FINALIZED if complete, INVALID if failed validation
+  ```
+- Expected errors documentation is mandatory (see section `Error Documentation` below)
 
-     // Example 2 - Additional context needed:
-     // GetPipeline returns the execution pipeline, or nil if not configured.
+### Error Documentation
+There are 2 categories of returned errors:
+  - Expected benign errors
+  - Exceptions (unexpected errors)
+
+Expected errors are benign sentinel errors returned by the function.
+Exceptions are unexpected errors returned by the function. These include all errors that are not benign within the context of the method.
```

Comment:
this distinction helped the agents understand that some error paths do not return sentinels. before it assumed any error returned was a sentinel and it just needed to look hard enough to find them.